src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -238,6 +239,189 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
+                writer.Indent++;
+                writer.WriteLine();
+
+                string writerClass = "XmlSerializationWriter" + suffix;
+                writerClass = classes.AddUnique(writerClass, writerClass);
+                var writerCodeGen = new XmlSerializationWriterCodeGen(writer, scopes, "public", writerClass);
+                writerCodeGen.GenerateBegin();
+                string[] writeMethodNames = new string[xmlMappings.Length];
+
+                for (int i = 0; i < xmlMappings.Length; i++)
+                {
+                    writeMethodNames[i] = writerCodeGen.GenerateElement(xmlMappings[i]);
+                }
+
+                writerCodeGen.GenerateEnd();
+                writer.WriteLine();
+
+                string readerClass = "XmlSerializationReader" + suffix;
+                readerClass = classes.AddUnique(readerClass, readerClass);
+                var readerCodeGen = new XmlSerializationReaderCodeGen(writer, scopes, "public", readerClass);
+                readerCodeGen.GenerateBegin();
+                string[] readMethodNames = new string[xmlMappings.Length];
+                for (int i = 0; i < xmlMappings.Length; i++)
+                {
+                    readMethodNames[i] = readerCodeGen.GenerateElement(xmlMappings[i]);
+                }
+
+                readerCodeGen.GenerateEnd(readMethodNames, xmlMappings, types);
+
+                string baseSerializer = readerCodeGen.GenerateBaseSerializer("XmlSerializer1", readerClass, writerClass, classes);
+                var serializers = new Hashtable();
+                for (int i = 0; i < xmlMappings.Length; i++)
+                {
+                    if (serializers[xmlMappings[i].Key] == null)
+                    {
+                        serializers[xmlMappings[i].Key] = readerCodeGen.GenerateTypedSerializer(readMethodNames[i], writeMethodNames[i], xmlMappings[i], classes, baseSerializer, readerClass, writerClass);
+                    }
+                }
+
+                readerCodeGen.GenerateSerializerContract("XmlSerializerContract", xmlMappings, types, readerClass, readMethodNames, writerClass, writeMethodNames, serializers);
+                writer.Indent--;
+                writer.WriteLine("}");
+                string serializerName = XmlSerializer.GetXmlSerializerAssemblyName(types[0], null);
+                string location = Path.Combine(codePath, serializerName + ".cs");
+                if (File.Exists(location))
+                {
+                    File.Delete(location);
+                }
+
+                FileStream fs;
+                using (fs = File.Create(location))
+                {
+                    string codecontent = compiler.Source.ToString();
+                    Byte[] info = new UTF8Encoding(true).GetBytes(codecontent);
+                    fs.Write(info, 0, info.Length);
+                }
+
+                return fs;
```

Comment:
`fs` is used in the `using` statements above. The FileStream is already disposed at this point.